### PR TITLE
when the operation returns void, the result is nil and to_json_value …

### PR DESCRIPTION
**Pull request by planthaber**
…is not defiend.

### CODE REVIEW

File: lib/rock/webapp/tasks/api.rb [hunk 1]
```diff
@@ -321,7 +321,12 @@ def get_operation(name_service, name, operation_name)
                         
                         begin
                             params = obj["args"]
-                            return op.callop(*params).to_json_value(:special_float_values => :string);
+                            result=op.callop(*params)
+                            if defined? result.to_json_value
```
Comment (doudou) [inline lib/rock/webapp/tasks/api.rb]: if result.respond_to?(:to_json_value)
else
end

File: lib/rock/webapp/tasks/api.rb [hunk 2]
```diff
@@ -321,7 +321,12 @@ def get_operation(name_service, name, operation_name)
...
+                                return result.to_json_value(:special_float_values => :string);
```
Comment (doudou) [inline lib/rock/webapp/tasks/api.rb]: And while you're at it, use the `key: :arg` syntax instead of `:key => :arg`

**Comment by planthaber** [comment]
Force-pushed the use of respond_to and key: :arg syntax

File: lib/rock/webapp/tasks/api.rb [hunk 3]
```diff
@@ -25,9 +25,9 @@ def self.stream_async_data_to_websocket(env, data_source, count = Float::INFINIT
                     listener = data_source.on_raw_data do |sample|
                         result = nil
                         if binary
-                            result = Hash[:mode => :binary, :value => sample.to_json_value(:pack_simple_arrays => true, :special_float_values => :string)]
+                            result = Hash[:mode => :binary, :value => sample.to_json_value(:pack_simple_arrays => true, special_float_values: :string)]
```
Comment (doudou) [inline lib/rock/webapp/tasks/api.rb]: That's ... weird. Changing one parameter and not all the others.

Comment (planthaber) [inline lib/rock/webapp/tasks/api.rb]: Thats right, use search/replace...

**Comment by planthaber** [comment]
changed to key: :arg syntax completly for api.rb

**Pull request merged**